i3lock.c

```diff
@@ -1743,23 +1840,11 @@ int main(int argc, char *argv[]) {
                                  (uint32_t[]){XCB_EVENT_MASK_STRUCTURE_NOTIFY});
 
     init_colors_once();
-    if (verify_png_image(image_path)) {
-        /* Create a pixmap to render on, fill it with the background color */
-        img = cairo_image_surface_create_from_png(image_path);
-    } else if (file_is_jpg(image_path)) {
-        DEBUG("Image looks like a jpeg, decoding\n");
-        unsigned char* jpg_data = read_JPEG_file(image_path, &jpg_info);
-            if (jpg_data != NULL) {
-                img = cairo_image_surface_create_for_data(jpg_data,
-                        CAIRO_FORMAT_ARGB32, jpg_info.width, jpg_info.height,
-                        jpg_info.stride);
-            }
-    }
-    /* In case loading failed, we just pretend no -i was specified. */
-    if (img && cairo_surface_status(img) != CAIRO_STATUS_SUCCESS) {
-        fprintf(stderr, "Could not load image \"%s\": %s\n",
-                image_path, cairo_status_to_string(cairo_surface_status(img)));
-        img = NULL;
+    if (is_regular_file(image_path)) {
+        img = load_image(image_path);
+    } else if (image_path != NULL) {
+        /* Path to a directory is provided -> use slideshow mode */
+        load_slideshow_images(image_path);
```

Comment:
I think it makes more sense to nest it like:

```
if (image_path != NULL) {
    if (is_regular_file(image_path)) {
        img = load_image(image_path);
    } else {
        load_slideshow_images(image_path);
    }
}
```

Just since there'll need to be a null-check for either of them, really, so we may as well hoist that check out a bit so it makes a bit more sense.

Otherwise, it all looks good and I'll merge it with dev after that.

Don't worry about rebasing on master/dev as they are right now, since I'm pretty used to that by now.